rotkehlchen/externalapis/beaconchain/service.py

```diff
@@ -78,6 +83,16 @@ def _query(
         May raise:
         - RemoteError due to problems querying beaconcha.in API
         """
+        if self.is_rate_limited():
+            log.error(
+                f'Beaconchain is rate limited until {self.ratelimited_until} when processing '
+                f'{module=} {endpoint=} {encoded_args=} with {data=}',
+            )
+            raise RemoteError(
+                'Beaconchain is rate limited until '
+                f'{timestamp_to_iso8601(self.ratelimited_until)}. Check logs for more details',
+            )
+
```

Comment:
i think you're duplicating logs here since `RemoteError` is still handled and the error is logged.

Reply:
The remoteError is more human readable since it can get to the user. It happens in the background task for querying produced blocks for example. The log above is there to give more context if we need it in the logs

Reply:
But won't this spam the user if the RemotError is consumed all the time and shown in the frontend?